Packages/com.unity.inputsystem/InputSystem/Devices/Sensor.cs

```diff
@@ -77,13 +81,56 @@ public static FourCC kFormat
             get { return new FourCC('L', 'A', 'A', 'C'); }
         }
 
-        [InputControl] public Vector3 acceleration;
+        [InputControl(processors = "Sensor")] 
+        public Vector3 acceleration;
 
         public FourCC GetFormat()
         {
             return kFormat;
         }
     }
+    
+    public class SensorProcessor : IInputControlProcessor<Vector3>
+    {
+        public Vector3 Process(Vector3 value, InputControl control)
+        {
+            if (!InputConfiguration.CompensateSensorsForScreenOrientation)
+                return value;
+            
+            Quaternion rotation = Quaternion.identity;
+            switch (Screen.orientation)
+            {
+                case ScreenOrientation.PortraitUpsideDown: rotation = Quaternion.Euler(0, 0, 180); break;
+                case ScreenOrientation.LandscapeLeft: rotation = Quaternion.Euler(0, 0, 90); break;
+                case ScreenOrientation.LandscapeRight: rotation = Quaternion.Euler(0, 0, 270); break;
+            }
+            return rotation * value;
+        }
+    }
+    
+    public class SensorRotationProcessor : IInputControlProcessor<Quaternion>
```

Comment:
Should have better name. SensorRotationProcessor gives no indication of how it processes rotation. Also no indication how this is really different from the processor above. Recommend weaving part of the name of the configuration property into the names. Stuff that makes it clear one processor is compensating a direction vector for screen orientation and the other processor is compensating a rotation for screen orientation.

IMO this doesn't even need to be specific to sensors and wouldn't have to have 'sensor' in the name. Any direction or rotation vector could be compensated for device orientation.

Also makes me wonder whether 'device orientation' is a better term than 'screen orientation'?

Reply:
I am really bad with names, but I'll try. Also device orientation is a different thing compared to screen orientation, so better not to mix them